packages/flutter/test/cupertino/route_test.dart

```diff
@@ -2691,6 +2691,54 @@ void main() {
     await tester.pumpAndSettle();
   });
 
+  testWidgets('fullscreen routes do not transition previous route', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      CupertinoApp(
+        initialRoute: '/',
+        onGenerateRoute: (RouteSettings settings) {
+          if (settings.name == '/') {
+            return PageRouteBuilder<void>(
+              pageBuilder: (_, __, ___) {
+                return CupertinoPageScaffold(
+                  navigationBar: const CupertinoNavigationBar(
+                    middle: Text('Page 1'),
+                  ),
+                  child: Container()
+                );
+              },
+            );
+          }
+          return CupertinoPageRoute<void>(
+            builder: (_) {
+              return CupertinoPageScaffold(
+                navigationBar: const CupertinoNavigationBar(
+                  middle: Text('Page 2'),
+                ),
+                child: Container(),
+              );
+            },
+            fullscreenDialog: true,
+          );
+        },
+      ),
+    );
+
+    expect(find.text('Page 1'), findsOneWidget);
+    expect(find.text('Page 2'), findsNothing);
+
+    final double pageTitleDX = tester.getTopLeft(find.text('Page 1')).dx;
+
+    tester.state<NavigatorState>(find.byType(Navigator)).pushNamed('/next');
+    await tester.pump();
+    await tester.pump(const Duration(milliseconds: 100));
```

Comment:
Why are two `pump`s needed here? Sorry if this is just copying another test that I already asked about, this looks familiar but I can't remember.

Reply:
An extra pump before a pump with a duration is necessary to push the route and begin the animation.